docs/design_doc.md

```diff
@@ -0,0 +1,443 @@
+
+-   [ ] P0. _Visualization_: we are developing a napari plugin for interactive visualization of _SpatialData_ objects @: scverse/napari-spatialdata
+-   [ ] P0. _Raw data IO_: we are implementing readers for raw data of common spatial omics technologies @: scverse/spatialdata-io . The goal is to provide initial readers but we don't plan to perform long-term maintenance, community contribution will be required. We will curate a tabular graphical representation of the initially supported technologies.
+-   [ ] P1. _Image analysis_: Library to perform image analysis, wrapping common analysis library in python such as skimage. Deprecate such functionalities in Squidpy.
+-   [ ] P2. _Database_: Some form of update on released datasets with updated specs as development progresses.
+
+## Detailed description
+
+### Terminology
+
+_SpatialData_ is both the name of the Python library as well as of the in-memory python object `SpatialData`. To distinguish between the two, we use the _italics_ formatting for the _SpatialData_ library and the `code` formatting for the `SpatialData` object.
+
+**_SpatialData_**
+
+The _SpatialData_ library provides a set of specifications and in-memory representations for spatial omics datasets with the goal of unifying spatial omics pipelines and making raw and processed datasets interoperable with browser-based viewers. _SpatialData_ also provides basic operations to query and manipulate such data. The _SpatialData_ specs inherit the OME-NGFF specs in full, yet for the start it defines additional specifications in particular around data types and metadata. _SpatialData_ also implements Python objects to load, save, and interact with spatial data.
+
+**Elements**
+
+Elements are the building blocks of _SpatialData_ datasets. Each element represents a particular datatype (e.g., raster image, label image, expression table). SpatialData elements are not special classes, but are instead standard scientific Python classes (e.g., `xarray.DataArray`, `AnnData`) with specified metadata. The metadata provides the necessary information for displaying and integrating the different elements (e.g., coordinate systems and coordinate transformations). Elements can either be initialized with valid metadata from disk or from in memory objects (e.g., `numpy` arrays) via _SpatialData_ helper functions. See the Elements section below for a detailed description of the different types of elements.
+
+**`SpatialData`**
+
+The `SpatialData` object contains a set of Elements to be used for analysis. Elements contained within a `SpatialData` object must be able to share a single Region Table. Future work may extend `SpatialData` to allow multiple tables (see discussion [here](https://github.com/scverse/spatialdata/issues/43)). All Elements within a `SpatialData` object can be queried, selected, and saved via the `SpatialData` objects.
+
+**`NGFFStore`**
+
+_Note: previously we used the term `SpatialDataContainer`; the new naming is `NGFFStore`._
```

Comment:
I think we can remove this reference? since there isn't a "before" of this document imho.

Reply:
Good point. Done.